ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala

```diff
@@ -107,5 +100,87 @@ abstract class ContractDaoBenchmark extends OracleAround {
       .unsafeRunSync()
     assert(inserted == batchSize)
   }
+}
+
+trait BenchmarkDbConnection {
+  def connectToDb(): Unit
+  def createDbJdbcConfig: JdbcConfig
+  def cleanup(): Unit
+}
+
+trait OracleBenchmarkDbConn extends BenchmarkDbConnection with OracleAround {
+
+  private var user: User = _
+  private val disableContractPayloadIndexing = false
+
+  override def connectToDb() = {
+    connectToOracle()
+  }
 
+  override def createDbJdbcConfig: JdbcConfig = {
+    user = createNewRandomUser()
+    val cfg = JdbcConfig(
+      dbutils.JdbcConfig(
+        driver = "oracle.jdbc.OracleDriver",
+        url = oracleJdbcUrl,
+        user = user.name,
+        password = user.pwd,
+        poolSize = ConnectionPool.PoolSize.Integration,
+      ),
+      dbStartupMode = DbStartupMode.CreateOnly,
+      backendSpecificConf =
+        if (disableContractPayloadIndexing) Map(DisableContractPayloadIndexing -> "true")
+        else Map.empty,
+    )
+    cfg
+  }
+
+  override def cleanup(): Unit = {
+    dropUser(user.name)
+  }
+}
+
+/*
+ Running the PG benchmark requires us to explicitly set the PGPASSWORD env variable which is picked
+ by the `createdb` tool when creating relevant database e.g
+
+ $ PGPASSWORD=<password> bazel run //ledger-service/http-json:contractdao-bench -- -f 1 -i 1 -wi 1 -bm avgt \
+ "QueryPayloadBenchmarkPostgres" -p extraParties=1,10 -p extraPayloadValues=1,10,100 -p batchSize=1000
+ */
+trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround {
+
+  @volatile
+  private var database: PostgresDatabase = _
+
+  private val host = sys.env.getOrElse("POSTGRESQL_HOST", "localhost")
+  private val port = Port(sys.env.get("POSTGRESQL_PORT").map(_.toInt).getOrElse(5432))
+  private val user = sys.env.getOrElse("POSTGRESQL_USERNAME", "postgres")
+  private val password = sys.env.getOrElse(
+    "PGPASSWORD",
+    throw new IllegalArgumentException(
+      s"Missing required PGPASSWORD env value, needed to run `createdb` "
+    ),
+  )
+
+  override def connectToDb() = {
+    connectToSharedServer(host, port, user, password)
+    database = createNewRandomDatabase()
+  }
+
+  override def createDbJdbcConfig: JdbcConfig = {
+    JdbcConfig(
+      dbutils.JdbcConfig(
+        driver = "org.postgresql.Driver",
+        url = database.url,
+        user = user,
+        password = password,
+        poolSize = ConnectionPool.PoolSize.Integration,
```

Comment:
Since you need access to these, I would just make the read part of the `server` var visible:

```scala
  protected[this] def server = server_
  @volatile private var server_ : PostgresServer = _
```

and use the properties therein here.

Reply:
didn't need this change ,`PostgresDatabase` class exposes the necessary information